cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java

```diff
@@ -268,6 +269,51 @@ public void testCompatibility() {
     }
   }
 
+
+  @Test
+  public void testFromMetadataEntity() {
+    ApplicationId applicationId = new ApplicationId("testNs", "app1");
+    Assert.assertEquals(applicationId, EntityId.fromMetadataEntity(applicationId.toMetadataEntity()));
+
+    MetadataEntity metadataEntity = MetadataEntity.ofNamespace("testNs").appendAsType(MetadataEntity.APPLICATION,
+                                                                                      "app1");
+    Assert.assertEquals(applicationId, EntityId.fromMetadataEntity(metadataEntity));
+
+    metadataEntity = MetadataEntity.ofDataset("testNs", "testDs").appendAsType("field", "testField");
+    try {
+      EntityId.fromMetadataEntity(metadataEntity);
+      Assert.fail("Should have failed to get create an EntityId from MetadataEntity");
+    } catch (IllegalArgumentException e) {
+      // expected
+    }
+
+    ProgramId programId = new ProgramId(applicationId, ProgramType.WORKER, "testWorker");
+    Assert.assertEquals(programId, EntityId.fromMetadataEntity(programId.toMetadataEntity()));
+
+    ArtifactId artifactId = new ArtifactId("testNs", "testArtifact-1.0.0.jar");
+    Assert.assertEquals(artifactId, EntityId.fromMetadataEntity(artifactId.toMetadataEntity()));
+  }
+
+  @Test
+  public void testGetKnownParentEntityId() {
+    MetadataEntity metadataEntity = MetadataEntity.ofDataset("testNs", "testDs").appendAsType("field", "testField");
```

Comment:
test the case where there is not known parent.
test the case where this entity is a known entity.

Reply:
👍